### PR TITLE
Improve types

### PR DESCRIPTION
**Pull request by t3chguy**
---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -4130,7 +4130,7 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
     public setPowerLevel(
         roomId: string,
         userId: string | string[],
-        powerLevel: number,
+        powerLevel: number | undefined,
```
Comment (gsouquet) [inline src/client.ts:4133]: Wondering whether the following would be more adequate? Or whether it's better to be really explicit?

```suggestion
        powerLevel?: number,
```

**Review by gsouquet — APPROVED** [review]
Otherwise lgtm 👍

File: src/client.ts (hunk 1 above)
Comment (t3chguy) [reply src/client.ts:4133]: Can't because the next field isn't optional, optional fields cannot be followed by mandatory fields

Comment (gsouquet) [reply src/client.ts:4133]: Woops, i read that wrong and thought we destructured the thing

**Pull request merged**